engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -116,6 +143,11 @@ public BlockRegion set(BlockRegion source) {
         return this;
     }
 
+    /**
+     * Create a fresh copy of this block region.
+     *
+     * @see BlockRegion#BlockRegion(BlockRegion)
+     */
     public BlockRegion copy() {
         return new BlockRegion(this);
     }
```

Comment:
this seems kind of redundant. do we need this?

Reply:
While writing tests I thought that I could read

```java
region.copy().doThis().doThat()
```

way better than
```java
new BlockRegion(region).doThis().doThat()
```

Might be just personal taste, but I thought this would not hurt...

Reply:
I'm guessing @pollend meant the comment rather than the method itself.
In order to get this in so we can align the other joml PRs with it and make progress, I'll dismiss your review @pollend but please let's talk about this on Discord if you want and if applicable do a follow-up PR.

Reply:
It's not too bad, I think it should be fine as is then. Just don't want there to be too many variations. Some of the convenience ends up being more confusing. might not be recognizable that copy means producing a new duplicate of the original. I don't think `new BlockRegion(a)` is any more convenient then `a.copy()`.  If we do something like then then be prepared to also do this for other situations. rather not having something purely unique only to BlockRegion if that makes sense.

Reply:
I actually did re-expose the constructors for min/max and single block position, as well as the copy-constructor taking another region. What's still left on the utility class are constructors for `fromMinAndSize` and `fromCenterAndExtents`. 

I can remove the `copy` method, but then I'd also go for removing the utility class, as
```java
fromMinAndSize(min, size) == new BlockRegion(min).setSize(size);

fromCenterAndExtents(center, extents) == new BlockRegion(center).expand(extents);
```

It would be nice to make `BlockRegion implements Iterable`, but then we'd need to throw out on of the iterables, too (preferably keep in the in-place variant, as you can implement the other on caller side).